### ISSUE TITLE
Allow indented property index?

### ISSUE THREAD

**Issue by helixbass** [jashkenas/coffeescript#5168]
@GeoffreyBooth I came across this while working on the Prettier formatter

Seems like this should be allowed:
```
a[
  if b
    c
  else
    d
]
```
But currently doesn't parse. I guess the current workaround would be to wrap in parens eg
```
a[(
  if b
    c
  else
    d
)]
```

Simple case:
```
a[
  b
]
```

**Comment by GeoffreyBooth** [comment]
Within an array block, a new line is supposed to signify a new array element. Taking advantage of line continuations here to make that `if`/`else` block one big element feels like a loophole.

For Prettier at least I would hope that it doesn’t encourage such output; there _should_ be parentheses or some other obvious indicator of where one element starts and ends, if they aren’t just one per line.

**Comment by helixbass** [comment]
@GeoffreyBooth this isn't about array indexes but rather property indexes, eg `a[b]`. So it's not about distinguishing multiple elements, just whether it's permissible to indent the index expression

You could say this should be allowed since in most (if not all?) other cases you can indent within a pair of explicit braces/parens/curly brackets, eg:
```
a(b)

a(
  b
)
```

```
(a) ->

(
  a
) ->
```

```
[a]

[
  a
]
```

```
{a}

{
  a
}
```

**Comment by GeoffreyBooth** [comment]
Ah sorry, disregard 😀

I still feel like we should be using Prettier more for reformatting code style and less for keeping within a line length. The hacks to get CoffeeScript under 80 or so characters per line generate some very ugly, ill-advised syntax. There are so many bugs around implicit line continuation that it's one of the things that _does_ often change between releases, breaking people who rely on its quirks; I wouldn't want there to be many many more such people if Prettier starts relying on line continuation quirks.

**Comment by GeoffreyBooth** [comment]
Fixed in #5193.
